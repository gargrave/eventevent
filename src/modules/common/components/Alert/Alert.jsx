// @flow
import React from 'react';
import { any, oneOf, string } from 'prop-types';

type Props = {
  children?: any,
  message: string,
  type: string,
};

const Alert = ({
  children,
  message,
  type,
}: Props) => (
    <div className={`alert alert-${type}`}>
      {message}
      {children}
    </div>
  );

Alert.propTypes = {
  children: any,
  message: string.isRequired,
  type: oneOf([
    'primary',
    'secondary',
    'success',
    'info',
    'warning',
    'danger',
    'light',
    'dark',
  ]).isRequired,
};

export default Alert;
