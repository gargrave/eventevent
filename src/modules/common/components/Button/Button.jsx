// @flow
import React from 'react';
import { bool, func, oneOf, string } from 'prop-types';

type Props = {
  canSubmit?: boolean,
  classes?: string,
  disabled?: boolean,
  onClick: Function,
  text: string,
  type?: string,
};

const Button = ({
  canSubmit,
  disabled,
  onClick,
  text,
  type = 'primary',
}: Props) => {
  return (
    <button
      className={`button button-${type}`}
      disabled={disabled || false}
      onClick={onClick}
      type={canSubmit ? 'submit' : 'button'}
    >
      {text}
    </button>
  );
};

Button.propTypes = {
  canSubmit: bool,
  classes: string,
  disabled: bool,
  onClick: func.isRequired,
  text: string.isRequired,
  type: oneOf([
    'primary',
    'success',
    'secondary',
    'info',
    'warning',
    'danger',
    'light',
    'dark',
  ]),
};

export default Button;
