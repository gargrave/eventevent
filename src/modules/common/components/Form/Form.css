.form {
  margin: auto;
  margin-top: 20px;
  max-width: 600px;
  text-align: left;
}
