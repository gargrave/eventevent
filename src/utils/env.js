const ENV = {
  DEV: 'development',
  PROD: 'production',
};
const DEV_ENV_URL = 'localhost:3000';

let currentEnv;
(function setCurrentEnv() {
  if (!currentEnv) {
    const loc = window.location.href;
    if (loc.match(DEV_ENV_URL)) {
      currentEnv = ENV.DEV;
    } else {
      currentEnv = ENV.PROD;
    }
  }
})();

export const isDev = () => currentEnv === ENV.DEV;
export const isProd = () => currentEnv === ENV.PROD;
