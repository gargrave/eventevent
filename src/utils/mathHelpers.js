export const clamp = (val, min, max) => {
  if (val < min) {
    return min;
  }
  if (val > max) {
    return max;
  }
  return val;
};
