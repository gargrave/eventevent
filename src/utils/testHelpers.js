import React from 'react';
import { mount, shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = { ...defaultProps };
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }

  mountComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return mount(<this.component {...props} />);
  }
}
