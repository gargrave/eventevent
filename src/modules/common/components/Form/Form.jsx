// @flow
import React from 'react';
import { any, func, bool, string} from 'prop-types';

import Alert from '../Alert/Alert';
import Button from '../Button/Button';

import styles from './Form.css';

type Props = {
  cancelBtnText?: string,
  children?: any,
  disabled?: boolean,
  onCancel?: Function,
  onSubmit: Function,
  submitBtnText?: string,
  submitDisabled?: boolean,
  topLevelError?: string,
};

const Form = ({
  cancelBtnText,
  children,
  disabled,
  onCancel,
  onSubmit,
  submitBtnText,
  submitDisabled = false,
  topLevelError,
}: Props) => {
  return (
    <div>
      {topLevelError &&
        <Alert
          message={topLevelError}
          type="danger"
        />
      }
      <form
        className={styles.form}
        onSubmit={onSubmit}
        noValidate>

        {children}

        <div className="input-field">
          <Button
            canSubmit={true}
            disabled={submitDisabled || disabled || false}
            onClick={onSubmit}
            position="left"
            text={submitBtnText || 'Submit'}
            type="success"
          />

          {onCancel &&
            <Button
              onClick={onCancel}
              text={cancelBtnText || 'Cancel'}
              type="light"
            />
          }
        </div>
      </form>
    </div>
  );
};

Form.propTypes = {
  cancelBtnText: string,
  children: any,
  disabled: bool,
  onCancel: func,
  onSubmit: func.isRequired,
  submitBtnText: string,
  submitDisabled: bool,
  topLevelError: string,
};

export default Form;
