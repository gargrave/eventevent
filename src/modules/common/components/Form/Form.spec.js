import { ComponentBuilder } from '../../../../utils/testHelpers';

import Alert from '../Alert/Alert';
import Button from '../Button/Button';
import Form from './Form';

const defaultProps = () => ({
  cancelBtnText: 'Cancel',
  children: [],
  disabled: false,
  onCancel: jest.fn(),
  onSubmit: jest.fn(),
  submitBtnText: 'Submit',
  submitDisabled: false,
  topLevelError: '',
});

const builder = new ComponentBuilder(Form, defaultProps());

describe('Form', () => {
  let component;

  test('matches the snapshot', () => {
    component = builder.shallowGetComponent();
    expect(component).toMatchSnapshot();
  });

  describe('basic rendering', () => {
    test('renders  correctly', () => {
      component = builder.shallowGetComponent();
      expect(component.find(Alert)).toHaveLength(0);
      expect(component.find(Button)).toHaveLength(2);
    });

    test('does not render "cancel" button if prop is empty', () => {
      component = builder.shallowGetComponent({ onCancel: null });
      expect(component.find(Button)).toHaveLength(1);
    });

    test('renders an Alert if the prop is present', () => {
      component = builder.shallowGetComponent({ topLevelError: 'OMFG' });
      expect(component.find(Alert)).toHaveLength(1);
    });
  });
});
