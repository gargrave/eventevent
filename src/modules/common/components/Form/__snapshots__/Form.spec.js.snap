// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    cancelBtnText="Cancel"
    disabled={false}
    onCancel={[Function]}
    onSubmit={[Function]}
    submitBtnText="Submit"
    submitDisabled={false}
    topLevelError=""
>
    
</Form>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "",
        <form
          className={undefined}
          noValidate={true}
          onSubmit={[Function]}
>
          <div
                    className="input-field"
          >
                    <Button
                              canSubmit={true}
                              disabled={false}
                              onClick={[Function]}
                              position="left"
                              text="Submit"
                              type="success"
                    />
                    <Button
                              onClick={[Function]}
                              text="Cancel"
                              type="light"
                    />
          </div>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [],
            <div
              className="input-field"
>
              <Button
                            canSubmit={true}
                            disabled={false}
                            onClick={[Function]}
                            position="left"
                            text="Submit"
                            type="success"
              />
              <Button
                            onClick={[Function]}
                            text="Cancel"
                            type="light"
              />
</div>,
          ],
          "className": undefined,
          "noValidate": true,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Button
                  canSubmit={true}
                  disabled={false}
                  onClick={[Function]}
                  position="left"
                  text="Submit"
                  type="success"
/>,
                <Button
                  onClick={[Function]}
                  text="Cancel"
                  type="light"
/>,
              ],
              "className": "input-field",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "canSubmit": true,
                  "disabled": false,
                  "onClick": [Function],
                  "position": "left",
                  "text": "Submit",
                  "type": "success",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "onClick": [Function],
                  "text": "Cancel",
                  "type": "light",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <form
            className={undefined}
            noValidate={true}
            onSubmit={[Function]}
>
            <div
                        className="input-field"
            >
                        <Button
                                    canSubmit={true}
                                    disabled={false}
                                    onClick={[Function]}
                                    position="left"
                                    text="Submit"
                                    type="success"
                        />
                        <Button
                                    onClick={[Function]}
                                    text="Cancel"
                                    type="light"
                        />
            </div>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [],
              <div
                className="input-field"
>
                <Button
                                canSubmit={true}
                                disabled={false}
                                onClick={[Function]}
                                position="left"
                                text="Submit"
                                type="success"
                />
                <Button
                                onClick={[Function]}
                                text="Cancel"
                                type="light"
                />
</div>,
            ],
            "className": undefined,
            "noValidate": true,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Button
                    canSubmit={true}
                    disabled={false}
                    onClick={[Function]}
                    position="left"
                    text="Submit"
                    type="success"
/>,
                  <Button
                    onClick={[Function]}
                    text="Cancel"
                    type="light"
/>,
                ],
                "className": "input-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "canSubmit": true,
                    "disabled": false,
                    "onClick": [Function],
                    "position": "left",
                    "text": "Submit",
                    "type": "success",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "onClick": [Function],
                    "text": "Cancel",
                    "type": "light",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
